module/dkg/broker.go

```diff
@@ -415,31 +415,36 @@ func (b *Broker) listen() {
 // onPrivateMessage verifies the integrity of an incoming message and forwards
 // it to consumers via the msgCh.
 func (b *Broker) onPrivateMessage(originID flow.Identifier, msg messages.DKGMessage) {
-	err := b.checkMessageInstanceAndOrigin(msg)
-	if err != nil {
-		b.log.Err(err).Msg("bad message")
-		return
-	}
-	// check that the message's origin matches the sender's flow identifier
-	nodeID := b.committee[msg.Orig].NodeID
-	if !bytes.Equal(nodeID[:], originID[:]) {
-		b.log.Error().Msgf("bad message: OriginID (%v) does not match committee member %d (%v)", originID, msg.Orig, nodeID)
+	if memberIndex, ok := b.committee.GetIndex(originID); ok {
+		err := b.checkMessageInstanceAndOrigin(uint64(memberIndex), msg)
+		if err != nil {
+			b.log.Err(err).Msg("bad message")
+			return
+		}
+		// check that the message's origin matches the sender's flow identifier
+		nodeID := b.committee[memberIndex].NodeID
+		if !bytes.Equal(nodeID[:], originID[:]) {
+			b.log.Error().Msgf("bad message: OriginID (%v) does not match committee member %d (%v)", originID, memberIndex, nodeID)
+			return
+		}
```

Comment:
Not new code here, but I don't think this check is necessary besides as a sanity check against internal bugs. Since we are reading `memberIndex` from `committee`, this can only happen if `GetIndex` has a bug. I'm inclined to remove the check, but let me know what you think.

Reply:
I agree I wanted to remove it but left it just in case in what was there for some reason I could not see.

Reply:
https://github.com/onflow/flow-go/pull/2238/commits/7a287cfa9c3d71c3b4f593d01f47ba48e182dd29